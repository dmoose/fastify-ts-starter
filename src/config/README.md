# Config Folder

Configuration is handled by mozilla's [convict](https://www.npmjs.com/package/convict) package

The index file includes additional files based on the environment. Those json files are in this directory.

Convict uses environment and command line arguments to override the defaults provided by the schema. It also validates the final configuration via the schema.
