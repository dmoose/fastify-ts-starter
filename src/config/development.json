{
  "port": 3030,
  "db": {
    "host": "dev.db"
  }
}
