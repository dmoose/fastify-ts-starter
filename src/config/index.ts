/*
 * create the configuration with defaults
 * https://www.npmjs.com/package/convict
 */

import path from 'path'
import convict from 'convict'
import convict_format_with_validator = require('convict-format-with-validator')

convict.addFormat(convict_format_with_validator.ipaddress)

const config = convict({
  env: {
    doc: 'The application environment.',
    format: ['production', 'development', 'test'],
    default: 'development',
    env: 'NODE_ENV',
    arg: 'env',
  },
  ip: {
    doc: 'The IP address to bind.',
    format: 'ipaddress',
    default: '0.0.0.0',
    env: 'BIND_ADDRESS',
  },
  port: {
    doc: 'The port to bind.',
    format: 'port',
    default: 80,
    env: 'HTTP_PORT',
    arg: 'port',
  },
  db: {
    host: {
      doc: 'Database host name/IP',
      format: '*',
      default: 'server1.dev.test',
    },
    name: {
      doc: 'Database name',
      format: String,
      default: 'users',
    },
  },
})

// load environment dependent configuration
const env = config.get('env')
config.loadFile(path.join(__dirname, `${env}.json`))

config.validate({ allowed: 'strict' })
export default config
