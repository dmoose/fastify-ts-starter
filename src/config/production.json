{
  "port": 80,
  "db": {
    "host": "production.db"
  }
}
