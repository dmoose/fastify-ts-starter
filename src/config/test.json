{
  "port": 80,
  "db": {
    "host": "test.db"
  }
}
