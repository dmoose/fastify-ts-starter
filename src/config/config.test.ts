import config from '.'

const c = config.toString()

test('config is created correctly', () => {
  expect(c).toMatchSnapshot()
})
