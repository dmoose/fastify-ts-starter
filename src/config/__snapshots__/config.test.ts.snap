// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`config is created correctly 1`] = `
"{
  \\"env\\": \\"test\\",
  \\"ip\\": \\"0.0.0.0\\",
  \\"port\\": 80,
  \\"db\\": {
    \\"host\\": \\"test.db\\",
    \\"name\\": \\"users\\"
  }
}"
`;
